Sources/ClientRuntime/Networking/HashFunction.swift

```diff
@@ -69,6 +91,43 @@ enum HashFunction {
     }
 }
 
+extension HashFunction: Comparable {
+    /*
+     * Priority-order for validating checksum = [ CRC32C, CRC32, SHA1, SHA256 ]
+     * Order is determined by speed of the algorithm's implementation
+     * MD5 is not supported by list ordering
+     */
+    public static func < (lhs: HashFunction, rhs: HashFunction) -> Bool {
+        let order: [HashFunction] = [.crc32c, .crc32, .sha1, .sha256]
+
+        let lhsIndex = order.firstIndex(of: lhs) ?? Int.max
+        let rhsIndex = order.firstIndex(of: rhs) ?? Int.max
```

Comment:
So `.md5` is always last when compared to an algorithm listed on L101?

Reply:
md5 is not a valid flexible checksum as per the SEP so it will never appear as part of the requestAlgorithmMember. If flexible checksums workflow is skipped it should be the default used

Reply:
Adding an update to skip Content-MD5 calculation when using checksums, as per the SEP